chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PrecacheableDimensionalDataManagerImpl.java

```diff
@@ -0,0 +1,214 @@
+package gov.healthit.chpl.manager.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.caching.CacheNames;
+import gov.healthit.chpl.dao.CQMCriterionDAO;
+import gov.healthit.chpl.dao.CertificationCriterionDAO;
+import gov.healthit.chpl.dao.CertificationEditionDAO;
+import gov.healthit.chpl.dao.CertificationStatusDAO;
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.dao.PracticeTypeDAO;
+import gov.healthit.chpl.dao.ProductClassificationTypeDAO;
+import gov.healthit.chpl.dao.ProductDAO;
+import gov.healthit.chpl.domain.DescriptiveModel;
+import gov.healthit.chpl.domain.KeyValueModel;
+import gov.healthit.chpl.domain.KeyValueModelStatuses;
+import gov.healthit.chpl.dto.CQMCriterionDTO;
+import gov.healthit.chpl.dto.CertificationCriterionDTO;
+import gov.healthit.chpl.dto.CertificationEditionDTO;
+import gov.healthit.chpl.dto.CertificationStatusDTO;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.dto.PracticeTypeDTO;
+import gov.healthit.chpl.dto.ProductClassificationTypeDTO;
+import gov.healthit.chpl.dto.ProductDTO;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.manager.PrecacheableDimensionalDataManager;
+
+@Service("precacheableDimensionalDataManager")
+public class PrecacheableDimensionalDataManagerImpl implements PrecacheableDimensionalDataManager {
+    private static final Logger LOGGER = LogManager.getLogger(PrecacheableDimensionalDataManagerImpl.class);
+    @Autowired
```

Comment:
We've tried to move these autowired things to constuctors. Can this do that too?